cmake/packaging.cmake

```diff
@@ -19,36 +19,36 @@ set(CPACK_PACKAGE_HOMEPAGE_URL "https://github.com/lifting-bits/sleigh")
 set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${PROJECT_VERSION}-${PACKAGE_VERSION}.x86_64")
 set(CPACK_PACKAGE_RELOCATABLE ON)
 
-set(CPACK_DEBIAN_PACKAGE_RELEASE "${PACKAGE_VERSION}")
-set(CPACK_DEBIAN_PACKAGE_PRIORITY "extra")
-set(CPACK_DEBIAN_PACKAGE_SECTION "default")
-set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "${CPACK_PACKAGE_HOMEPAGE_URL}")
-set(CPACK_DEBIAN_PACKAGE_DEPENDS "libc-dev-bin (>=2.31)")
-
-set(CPACK_RPM_PACKAGE_RELEASE_DIST "${PACKAGE_VERSION}")
-set(CPACK_RPM_PACKAGE_DESCRIPTION "${CPACK_PACKAGE_DESCRIPTION_SUMMARY}")
-set(CPACK_RPM_PACKAGE_GROUP "default")
-set(CPACK_RPM_PACKAGE_REQUIRES "glibc >= 2.31")
-
 set(CPACK_GENERATOR "TGZ")
 
-unset(rpm_executable_path CACHE)
-find_program(rpm_executable_path "rpm")
-if("${rpm_executable_path}" STREQUAL "rpm_executable_path-NOTFOUND")
-  message(WARNING "sleigh: the RPM package generator requires the 'rpm' tool")
-
-else()
-  list(APPEND CPACK_GENERATOR "RPM")
-  message(STATUS "sleigh: the RPM generator has been enabled")
-endif()
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
```

Comment:
Am I reading this correctly that RPMs only get built on Linux? `rpm` is available via Homebrew so I think we should check for "not Windows".

Reply:
I've honestly never heard of someone using rpm to manage their macOS package installations. I thought the `rpm` Hombrew formula was more for supporting cross-compilation from macOS to Linux.

Also, the packaging metadata we specify is for Linux with glibc; macOS doesn't use glibc
https://github.com/lifting-bits/sleigh/blob/04db45f0b73372aa038e79b7e3fc44c3eb14732b/cmake/packaging.cmake#L36

Semi-related, I don't think the glibc version number is accurate because we don't really enforce that in CI when we build and publish the pre-built packages. Nor do I know if Sleigh actually requires a minimum specific glibc version for a particular feature.

We should probably delete the above line with glibc, but what do you think?

Same goes for the dpkg specification.

Reply:
> I've honestly never heard of someone using rpm to manage their macOS package installations. I thought the `rpm` Hombrew formula was more for supporting cross-compilation from macOS to Linux.

Fair enough, let's leave the check as is.

> We should probably delete the above line with glibc, but what do you think?

Hmm, I'm not sure where that came from originally. I agree though, if we don't know for sure let's not constrain it like this.

Reply:
Left the check as-is and removed the `PACKAGE_REQUIRES` lines.